service/shadowsocks.go

```diff
@@ -44,7 +44,7 @@ type ServiceMetrics interface {
 
 type Service interface {
 	HandleStream(ctx context.Context, conn transport.StreamConn)
-	HandlePacket(conn net.PacketConn)
+	HandlePacket(conn net.Conn, pkt []byte)
```

Comment:
The API is also unclear. Why does it need a full net.Conn when it already has the packet?